### ISSUE TITLE
VHPOP-2.2 does not compile with newer version of GCC

### ISSUE THREAD

**Issue by GoogleCodeExporter** [hlsyounes/vhpop#2]
```
What steps will reproduce the problem?
1. Downloading vhpop-2.2.tar.gz_FILES and unzipping it
2. type ./configure in the terminal
3. type make

What is the expected output? What do you see instead?
The code should get installed without errors. But what I see is as follows:

source='vhpop.cc' object='vhpop.o' libtool=no \
depfile='.deps/vhpop.Po' tmpdepfile='.deps/vhpop.TPo' \
depmode=gcc3 /bin/sh ./depcomp \
g++ -DHAVE_CONFIG_H -I. -I. -I.     -g -O2 -c -o vhpop.o `test -f 'vhpop.cc' || 
echo './'`vhpop.cc
hashing.h:55: error: ‘hash’ is not a template
orderings.h:159: error: ‘hashing::hash_map’ has not been declared
orderings.h:159: error: expected ‘,’ or ‘...’ before ‘<’ token
orderings.h:164: error: expected template-name before ‘<’ token
orderings.h:164: error: expected `{' before ‘<’ token
orderings.h:164: error: expected unqualified-id before ‘<’ token
orderings.h:168: error: ‘const_iterator’ in class ‘Orderings::IdMap’ 
does not name a type
orderings.h:171: error: field ‘id_map1_’ has incomplete type
orderings.h:189: error: ‘hashing::hash_map’ has not been declared
orderings.h:189: error: expected ‘,’ or ‘...’ before ‘<’ token
orderings.h:191: error: ‘virtual float Orderings::schedule(int) const’ and 
‘float Orderings::schedule(int) const’ cannot be overloaded
orderings.h:118: confused by earlier errors, bailing out
make: *** [vhpop.o] Error 1
[mast@~/Desktop/RIP1/vhpop-2.2.tar.gz_FILES/vhpop-2.2]$ make clean
test -z "vhpop" || rm -f vhpop
rm -f *.o core *.core

What version of the product are you using? On what operating system?
vhpop-2.2

Please provide any additional information below.
I am using it on Fedora Core 4

```

---

Original issue reported on code.google.com by zafarmun...@gmail.com on 19 Sep 2011 at 5:46

**Comment by GoogleCodeExporter** [comment]
```
VHPOP 2.2 and earlier used the non-standard hash_map container.

Your best option is to get the source code for VHPOP from trunk (unofficially 
known as VHPOP 3.0):

$ svn checkout http://ymer.googlecode.com/svn/trunk/ ymer-read-only

Let me know if that works for you?

Can you tell me what version of GCC you are using?  When I try to compile with 
GCC 4.4, I get a different error:

source='vhpop.cc' object='vhpop.o' libtool=no \
    depfile='.deps/vhpop.Po' tmpdepfile='.deps/vhpop.TPo' \
    depmode=gcc3 /bin/sh ./depcomp \
    g++ -DHAVE_CONFIG_H -I. -I. -I.     -g -O2 -c -o vhpop.o `test -f 'vhpop.cc' || echo './'`vhpop.cc
In file included from /usr/include/c++/4.4/ext/hash_map:60,
                 from hashing.h:34,
                 from orderings.h:26,
                 from plans.h:27,
                 from vhpop.cc:20:
/usr/include/c++/4.4/backward/backward_warning.h:28:2: warning: #warning This 
file includes at least one deprecated or antiquated header which may be removed 
without further notice at a future date. Please use a non-deprecated interface 
with equivalent functionality instead. For a listing of replacement headers and 
interfaces, consult the file backward_warning.h. To disable this warning use 
-Wno-deprecated.
vhpop.cc: In function ‘bool read_file(const char*)’:
vhpop.cc:166: error: ‘strerror’ was not declared in this scope
vhpop.cc: In function ‘int main(int, char**)’:
vhpop.cc:299: error: ‘UINT_MAX’ was not declared in this scope
vhpop.cc:322: error: ‘UINT_MAX’ was not declared in this scope
make[1]: *** [vhpop.o] Error 1

```

---

Original comment by hlsyou...@gmail.com on 20 Sep 2011 at 5:16
- Changed title: **VHPOP-2.2 does not compile with newer version of GCC**
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r578.
```

---

Original comment by hlsyou...@gmail.com on 8 Nov 2013 at 7:57
- Changed state: **Fixed**
